### PR TITLE
fix acr build for pull request commit references

### PR DESCRIPTION
**Pull request by xyxyxyxyxyxy**
**Purpose of the PR**
Fix acr build for pull request commit references, e.g., https://github.com/northtyphoon/node-helloworld.git#pull/1/head
- 

Fixes #

### CODE REVIEW

File: scan/git.go [hunk 1]
```diff
@@ -150,7 +150,14 @@ func gitLfs(root string) error {
 func checkoutGit(root, ref, subdir string) (string, error) {
 	// Try checking out by ref name first. This will work on branches and sets
 	// .git/HEAD to the current branch name
-	if output, err := gitWithinDir(root, "checkout", ref); err != nil {
+	// If the reference format is "pull/{pull-request-number}/head", then checkout to
```
Comment (northtyphoon) [inline scan/git.go:153]: if the repo has multiple pull requests, which one does the code checkout?

Comment (northtyphoon) [reply scan/git.go:153]: Can you add a comments to clarify for pull request reference, previous step has already fetched it explicitly and current step just needs to check out the head.

Comment (xyxyxyxyxyxy) [reply scan/git.go:153]: Added

**Review by northtyphoon — APPROVED** [review]
LGTM

**Pull request merged**